### PR TITLE
Add support for gitpod

### PR DESCRIPTION
**Pull request by jlengrand**

### CODE REVIEW

**Comment by jlengrand** [comment]
The code now runs, but does not show in the preview. Not sure what is happening

**Comment by Kwok-he-Chu** [comment]
@jlengrand Did an upmerge with my latest changes, gitpod should work now. We're using the port :8080 for the application so it's consistent with the other samples that we have.

File: .gitpod.Dockerfile [hunk 1]
```diff
@@ -0,0 +1 @@
+FROM gitpod/workspace-dotnet:latest
```
Comment (jlengrand) [inline .gitpod.Dockerfile:1]: Not used any more ?

File: Controllers/HomeController.cs [hunk 2]
```diff
@@ -15,31 +15,35 @@ public HomeController(ILogger<HomeController> logger)
             _client_key = Environment.GetEnvironmentVariable("ADYEN_CLIENT_KEY");
         }
 
+        [Route("/")]
         public IActionResult Index()
         {
             return View();
         }
 
+        [Route("preview/{id}")]
```
Comment (jlengrand) [inline Controllers/HomeController.cs:24]: Nice, didn't know about that :)

File: Controllers/HomeController.cs [hunk 3]
```diff
@@ -71,6 +75,7 @@ public IActionResult Result(string status, [FromQuery(Name = "reason")] string r
         }
 
         [ResponseCache(Duration = 0, Location = ResponseCacheLocation.None, NoStore = true)]
+        [Route("error")]
```
Comment (jlengrand) [inline Controllers/HomeController.cs:78]: I think we use /result/error in other samples ? https://github.com/adyen-examples/adyen-node-online-payments/blob/main/index.js

**Comment by jlengrand** [comment]
Did you try Gitpod @Kwok-he-Chu ? It works much better than before indeed, but if I try the final redirection fails with gitpod. Is it the same for you?

File: .gitpod.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+#image:
+#  file: .gitpod.Dockerfile
```
[the remark this answers is not in the document]
Comment (Kwok-he-Chu) [reply .gitpod.yml:2]: Removed

File: .gitpod.Dockerfile (hunk 1 above)
Comment (Kwok-he-Chu) [reply .gitpod.Dockerfile:1]: Will remove

File: Controllers/HomeController.cs [hunk 5]
```diff
@@ -15,31 +15,35 @@ public HomeController(ILogger<HomeController> logger)
...
+
+        [Route("redirect")]
         public IActionResult Redirect()
         {
             ViewBag.ClientKey = _client_key;
             return View();
         }
 
-        [HttpGet("Home/Result/{status}")]
+        [HttpGet("Home/result/{status}")]
```
[the remark this answers is not in the document]
Comment (Kwok-he-Chu) [reply Controllers/HomeController.cs:46]: Good notice! Forgot to remove this particular line!

File: Controllers/HomeController.cs (hunk 3 above)
Comment (Kwok-he-Chu) [reply Controllers/HomeController.cs:78]: This is the default error page. The /result/error page is part of the [HttpGet("/result/{status}")] function, which is resolved properly now 👍

**Comment by Kwok-he-Chu** [comment]
> Did you try Gitpod @Kwok-he-Chu ? It works much better than before indeed, but if I try the final redirection fails with gitpod. Is it the same for you?

I've tried every Drop-in method/Card in gitpod and it works. I've only seen it go wrong for iDeal where the returnUrl points to localhost:8080 on the iDeal simulation screen. The redirect will then fail since it will point to localhost:8080. A user would have to manually paste the correct returnUrl. I've seen the same behavior with the python library actually. I'm thinking whether we should fix it or leave it with a README note, what is your opinion on this?

**Comment by jlengrand** [comment]
Great. So in some of the samples (we want all actually), we do that dynamically now : https://github.com/adyen-examples/adyen-java-spring-online-payments/blob/main/src/main/java/com/adyen/checkout/api/CheckoutResource.java#L59.

Coudl be an idea

**Comment by Kwok-he-Chu** [comment]
@jlengrand Made the change to retrieve the host url from the headers origin properly, please review/test 👍 Should work

**Pull request merged**